pkg/service/psetupsvc/complete_step.go

```diff
@@ -80,7 +82,7 @@ func (s *Service) CompleteStep(ctx context.Context, setupID string, stepIndex in
 
 		// any secret starting with 'awsssm://' is assumed to be an existing reference
 		// to a secret and is not set.
-		if strings.HasPrefix(value, "awsssm://") {
+		if f.IsSecret() && strings.HasPrefix(value, "awsssm://") {
```

Comment:
Adding this additional constraint here as it should really only apply to secret fields